posts/polyglot.md

```diff
@@ -0,0 +1,109 @@
+
+These differences in syntax can be hard on the eyes and cause deterioration in mental focus, especially when the language is extremely unfamiliar.  
+
+### Standard Library
+
+All languages come with a standard library and very rarely are these even remotely similar from one language to another.
+
+Every application needs to make some use of standard library operations. As examples: How do you concatenate strings? How do you open a file? What is the best way to create a JSON string?
+
+Learning your way around these core operations is the second thing I find myself learning after picking up the necessary syntax.
+
+### Paradigm
+
+Ruby, Java, Python and C# share more in common with each other than they do with Erlang, Elixir, Haskell and Prolog.
+
+There are languages which straddle these paradigm borders, for example, OCaml and ECMAScript.
+
+It's important to be well versed in the reasons these languages are different and embrace their strengths, and avoid their weaknesses. In a language which predominantly favours writing in an Object-Oriented style should be used as an Object-Oriented language, rather than attempting to go against the grain and using potentially more esoteric language features. 
+
+Remember; as a polyglot, your code should be understandable by a monoglot programmer that *only* knows the language you are working in currently.
+
+### Culture
+
+Every language, and every stack within a language, and every paradigm approach applied to a stack and language has a sub-culture and associated idiosyncrasies. 
+
+As a polyglot, it is more important to be idiomatic within a sub-culture, than it is to attempt to harmonise the various quirks between the languages that you are familiar with personally. 
+
+If you are working with a team on a project which has a particular culture and approach to their code, you should attempt to work within the style that the rest of the team finds comfortable. Only at this point is it wise to begin to influence changes, and gain support from the team to adopt those changes. 
+
+Learning new cultures is by far the trickiest and requires a high degree of humility.
+
+## My polyglot toolbox
+
+Remembering all the differences between languages is hard work. Being a "specialist" in a language requires serious dedication, internalising knowledge about every corner of a language requires a lot of practical experience.
+
+So how do I remember all of this? The secret is: **I don't**. 
+
+Here, I'll tell you how I get by without being a specialist.
+
+### Be familiar with the paradigms
+
+It's important to learn new languages. The biggest piece of advice to those looking to become a polyglot is to pick languages that are vastly different to what you currently find comfortable.
```

Comment:
Haha, I just read a whole section on all the problems.  You've anchored my view to be negative.